auth_matrix/admin.py

```diff
@@ -32,22 +36,23 @@ class Meta:
         )
         export_order = fields
 
-    def dehydrate(self, user):
-        data = super().dehydrate(user)
-        group_names = [group.name for group in Group.objects.all()]
-        for group_name in group_names:
-            data[group_name] = group_name in [group.name for group in user.groups.all()]
+    def after_export(self, queryset, data, *args, **kwargs):
+        group_names_subquery = (
+            Group.objects.filter(user=OuterRef("pk"))
+            .values("user")
+            .annotate(group_names=ArrayAgg("name"), default=Value([]))
+            .values("group_names")
+        )
+        queryset = queryset.annotate(group_names=Subquery(group_names_subquery))
+        for group_name in Group.objects.values_list("name", flat=True).order_by("name"):
+            column = [group_name in (user.group_names or []) for user in queryset]
+            data.append_col(column, header=group_name)
         return data
 
-    def export(self, queryset, *args, **kwargs):
-        dataset = super().export(queryset, *args, **kwargs)
-        group_names = [group.name for group in Group.objects.all()]
-        for group_name in group_names:
-            dataset.append_col(
-                [user.groups.filter(name=group_name).exists() for user in queryset],
-                header=group_name,
-            )
-        return dataset
+
+class UserExportMixin(ExportMixin):
+    resource_classes = (UserGroupResource,)
```

Comment:
I keep getting confused by the name of this resource, so perhaps renaming it to `UserWithGroupsResource` will make it more obvious that we are indeed dealing with a user resource (and then the queryset/model confusion from earlier should hopefully not happen again).